src/inmanta/env.py

```diff
@@ -508,6 +511,7 @@ def create_log_content_files(title: str, files: list[str]) -> list[str]:
             not_found: list[str] = []
             conflicts: list[str] = []
             indexes: str = ""
+            no_index: bool = "--no-index" in cmd or strtobool(env.get("PIP_NO_INDEX", "false").lower())
```

Comment:
We have to move this code into the `if not_found` block. Otherwise this code can raise a `ValueError` if the value of the environment variable is invalid. Once we enter the `if non_found` block, we know that the input was not rejected by pip and we know that there will be no `ValueError`. Let's also add a comment about this in the code.